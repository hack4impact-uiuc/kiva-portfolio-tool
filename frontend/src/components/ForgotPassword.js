import { Link } from 'react-router-dom'
import React, { Component } from 'react'
import {
  getSecurityQuestionForUser,
  submitSecurityQuestionAnswer,
  resetPassword
} from '../utils/ApiWrapper'
import { Form, Button, FormGroup, Label, Input, Card, Alert, CardBody, CardTitle } from 'reactstrap'
import { connect } from 'react-redux'
import { setCookie } from '../utils/cookie'
import Navbar from './NavBar'
import BackgroundSlideshow from 'react-background-slideshow'

import '../styles/index.css'
import '../styles/login.css'

import kivaLogo from '../media/kivaPlainLogo.png'
import b1 from '../media/b1-min.jpg'
import b3 from '../media/b3-min.jpg'
import b4 from '../media/b4-min.jpg'
import b5 from '../media/b5-min.jpg'
import b6 from '../media/b6-min.jpg'

const EMAIL_REGEX =
  "([a-zA-Z0-9!#$%&'*+/=?^_`{|}~-]+)@([a-zA-Z0-9!#$%&'*+/=?^_`{|}~-]+).([a-zA-Z]{2,3}).?([a-zA-Z]{0,3})"
// const PASSWORD_REGEX = "^(((?=.*[a-z])(?=.*[A-Z]))|((?=.*[a-z])(?=.*[0-9]))|((?=.*[A-Z])(?=.*[0-9])))(?=.{6,})";

export class ForgotPassword extends Component {
  state = {
    email: '',
    question: '',
    errorMessage: '',
    answer: '',
    pin: '',
    password: '',
    password2: '',
    loadingAPI: false,
    submitNewPassword: false
  }

  handleChange = event => {
    this.setState({ [event.target.name]: event.target.value })
  }

  handleGetSecurityQuestion = async e => {
    e.preventDefault()
    const result = await getSecurityQuestionForUser(this.state.email)
    if (result) {
      const resp = await result.json()
      if (!!resp.question) {
        this.setState({ question: resp.question, errorMessage: '' })
      } else {
        this.setState({ errorMessage: resp.message })
      }
    }
  }

  handleSubmitSecurityAnswer = async e => {
    e.preventDefault()

    this.setState({ loadingAPI: true })
    const result = await submitSecurityQuestionAnswer(this.state.email, this.state.answer)

    if (result) {
      const resp = await result.json()
      if (resp.status === 200) {
        this.setState({ submitNewPassword: true, errorMessage: '' })
      } else {
        this.setState({ errorMessage: resp.message })
      }
    }
  }

  handleSubmitNewPassword = async e => {
    e.preventDefault()
    if (this.state.password !== this.state.password2) {
      this.setState({ errorMessage: "Passwords don't match!" })
      return
    }
    const response = await resetPassword(
      this.state.pin,
      this.state.email,
      this.state.password,
      this.state.answer
    )
    if (response) {
      response = response.json()
      if (response.status === 200 && response.token) {
        setCookie('token', response.token)
        this.setState({ successfulSubmit: true })
        this.props.history.push('/')
      } else {
        this.setState({ errorMessage: response.message })
      }
    }
  }

  render = () => (
    <div>
      <Navbar className="nav-absolute" />
      <div className="background">
        <BackgroundSlideshow images={[b1, b3, b4, b5, b6]} animationDelay={5000} />
      </div>
      {this.state.errorMessage !== '' && <Alert color="danger">{this.state.errorMessage}</Alert>}
      {this.state.submitNewPassword ? (
        <div className="foreground">
          <Card className="interview-card">
            <CardTitle>
              <div className="text-centered" id="login-kiva-logo">
                <img src={kivaLogo} />
              </div>
            </CardTitle>

            <CardBody>
              <Form>
                <FormGroup>
                  <Label>Pin</Label>
                  <Input name="pin" value={this.state.pin} onChange={this.handleChange} required />
                </FormGroup>
                <FormGroup>
                  <Input
                    type="password"
                    name="password"
                    placeholder="Password"
                    id="examplePassword"
                    minLength="8"
                    maxLength="64"
                    value={this.state.password}
                    onChange={this.handleChange}
                    required
                  />
                </FormGroup>
                <FormGroup>
                  <Input
                    type="password"
                    name="password2"
                    placeholder="Confirm Password"
                    id="exampleConfirm"
                    minLength="8"
                    maxLength="64"
                    value={this.state.password2}
                    onChange={this.handleChange}
                    required
                  />
                </FormGroup>
                <div className="text-centered">
                  <Button
                    color="success"
                    size="lg"
                    onClick={this.handleSubmitNewPassword}
                    className="right"
                  >
                    Reset Password
                  </Button>{' '}
                </div>
              </Form>
            </CardBody>
            <div style={{ textAlign: 'center' }}>
              <Link to="/login" prefetch href="/login">
                <a>Back to login page</a>
              </Link>
            </div>
          </Card>
        </div>
      ) : (
        <div className="foreground">
          {this.state.question === '' ? (
            <Card className="interview-card">
              <CardTitle>
                <div className="text-centered" id="login-kiva-logo">
                  <img src={kivaLogo} />
                </div>
              </CardTitle>

              <CardBody>
                <Form>
                  <FormGroup>
                    <Input
                      type="email"
                      name="email"
                      placeholder="E-mail"
                      id="exampleEmail"
                      maxLength="64"
                      pattern={EMAIL_REGEX}
                      value={this.state.email}
                      onChange={this.handleChange}
                      required
                    />
                  </FormGroup>
                  <div className="text-centered">
                    <Button
                      color="success"
                      size="lg"
                      onClick={this.handleGetSecurityQuestion}
                      className="right"
                    >
                      Get Security Question
                    </Button>

                    {this.state.errorMessage}
                  </div>
                </Form>
              </CardBody>
              <div style={{ textAlign: 'center' }}>
                <Link to="/login" prefetch href="/login">
                  <a>Back to login page</a>
                </Link>
              </div>
            </Card>
          ) : (
            <Card className="interview-card">
              <CardTitle>
                <div className="text-centered" id="login-kiva-logo">
                  <img src={kivaLogo} />
                </div>
              </CardTitle>

              <CardBody>
                <Form>
                  <FormGroup>
                    <p> {this.state.question}</p>
                    <Input
                      type="answer"
                      name="answer"
                      placeholder="Answer"
                      id="exampleAnswer"
                      onChange={this.handleChange}
                      required
                    />
                  </FormGroup>
                  <div className="text-centered">
                    <Button
                      color="success"
                      size="lg"
                      onClick={this.handleSubmitSecurityAnswer}
                      className="right"
                      disabled={this.state.loadingAPI}
                    >
                      Submit Answer
                    </Button>
                  </div>
                </Form>
              </CardBody>
              <div style={{ textAlign: 'center' }}>
                <Link to="/login" prefetch href="/login">
                  <a>Back to login page</a>
                </Link>
              </div>
            </Card>
          )}
        </div>
      )}
    </div>
  )
}
export default connect()(ForgotPassword)
