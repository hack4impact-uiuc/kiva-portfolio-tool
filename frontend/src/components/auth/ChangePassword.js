import React, { Component } from 'react'
import WithAuth from './WithAuth'
import Navbar from '../NavBar'
import { changePassword } from '../../utils/ApiWrapper'
import { Form, Button, FormGroup, Label, Input, CardTitle, Card, CardBody, Row } from 'reactstrap'
import { connect } from 'react-redux'
import { setCookie } from '../../utils/cookie'
import BackgroundSlideshow from 'react-background-slideshow'

import '../../styles/index.css'
import '../../styles/login.css'

import kivaLogo from '../../media/kivaPlainLogo.png'
import b1 from '../../media/b1-min.jpg'
import b3 from '../../media/b3-min.jpg'
import b4 from '../../media/b4-min.jpg'
import b5 from '../../media/b5-min.jpg'
import b6 from '../../media/b6-min.jpg'

class ChangePassword extends Component {
  state = {
    oldPassword: '',
    newPassword1: '',
    newPassword2: '',
    responseMessage: '',
    correctMessage: ''
  }

  handleChange = event => {
    this.setState({ [event.target.name]: event.target.value })
  }

  handlePassChange = async e => {
    e.preventDefault()
    if (this.state.newPassword1 === this.state.newPassword2) {
      const result = await changePassword(this.state.oldPassword, this.state.newPassword1)
      if (
        result.error != null &&
        (result.error.response.status === 400 || result.error.response.status === 500)
      ) {
        this.setState({
          responseMessage: result.error.response.data.message
        })
        return
      }

      let token = result.response.data.result.token
      setCookie('token', token)
      this.setState({ correctMessage: 'Your password has been changed!' })
    } else {
      this.setState({ responseMessage: 'Passwords do not match' })
    }
  }

  render() {
    return (
      <div>
        <Navbar className="nav-absolute" />
        <div className="background">
          <BackgroundSlideshow images={[b1, b3, b4, b5, b6]} animationDelay={5000} />
        </div>
        <Row>
          <div className="foreground">
            <Card className="interview-card">
              <CardTitle>
                <div className="text-centered" id="login-kiva-logo">
                  <img src={kivaLogo} alt="Kiva logo" />
                </div>
              </CardTitle>

              <CardBody>
                <Form>
                  <FormGroup>
                    <Input
                      name="oldPassword"
                      placeholder="Old Password"
                      type="password"
                      maxLength="128"
                      value={this.state.oldPassword}
                      onChange={this.handleChange}
                      required
                    />
                  </FormGroup>
                  <FormGroup>
                    <Input
                      name="newPassword1"
                      placeholder="New Password"
                      type="password"
                      maxLength="128"
                      value={this.state.newPassword1}
                      onChange={this.handleChange}
                      required
                    />
                  </FormGroup>
                  <FormGroup>
                    <Input
                      name="newPassword2"
                      placeholder="Confirm Password"
                      type="password"
                      maxLength="128"
                      value={this.state.newPassword2}
                      onChange={this.handleChange}
                      required
                    />
                  </FormGroup>
                  <Button
                    color="success"
                    size="lg"
                    onClick={this.handlePassChange}
                    className="left left-margin-lg"
                  >
                    Change Password
                  </Button>
                  <Button
                    color="success"
                    size="lg"
                    onClick={() => this.props.history.push('/')}
                    className="right"
                  >
                    Back To LogIn
                  </Button>
                </Form>
                <p style={{ color: 'red', textAlign: 'center' }}>
                  {this.state.responseMessage ? this.state.responseMessage : ''}
                </p>
                <p style={{ color: 'green', textAlign: 'center' }}>
                  {this.state.correctMessage ? this.state.correctMessage : ''}
                </p>
              </CardBody>
            </Card>
          </div>
        </Row>
      </div>
    )
  }
}

export default connect()(WithAuth(ChangePassword))
