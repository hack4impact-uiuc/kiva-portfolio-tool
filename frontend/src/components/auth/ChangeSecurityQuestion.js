import React, { Component } from 'react'
import WithAuth from './WithAuth'
import Navbar from '../NavBar'
import {
  changePassword,
  updateSecurityQuestion,
  getSecurityQuestions
} from '../../utils/ApiWrapper'
import {
  Form,
  Button,
  FormGroup,
  Label,
  Input,
  CardTitle,
  Card,
  CardBody,
  Row,
  Dropdown,
  DropdownItem,
  DropdownToggle,
  DropdownMenu
} from 'reactstrap'
import { connect } from 'react-redux'
import { setCookie } from '../../utils/cookie'
import BackgroundSlideshow from 'react-background-slideshow'

import '../../styles/login.scss'

import kivaLogo from '../../media/kivaPlainLogo.png'
import b1 from '../../media/b1-min.jpg'
import b3 from '../../media/b3-min.jpg'
import b4 from '../../media/b4-min.jpg'
import b5 from '../../media/b5-min.jpg'
import b6 from '../../media/b6-min.jpg'

const EMAIL_REGEX =
  "([a-zA-Z0-9!#$%&'*+/=?^_`{|}~-]+)@([a-zA-Z0-9!#$%&'*+/=?^_`{|}~-]+).([a-zA-Z]{2,3}).?([a-zA-Z]{0,3})"

class ChangePassword extends Component {
  state = {
    oldPassword: '',
    responseMessage: '',
    correctMessage: '',
    questions: [],
    questionIdx: -1,
    dropdownOpen: false,
    securityQuestionAnswer: '',
    failed: false,
    modal: false
  }

  pickDropDown = (idx, e) => {
    this.setState({ questionIdx: idx })
  }
  toggle = () => {
    this.setState({ dropdownOpen: !this.state.dropdownOpen })
  }

  handleChangeSecurityAnswer = event => {
    const value = event.target.value
    this.setState({ securityQuestionAnswer: value })
  }

  handleChange = event => {
    this.setState({ [event.target.name]: event.target.value })
  }

  async componentDidMount() {
    this.setState({ loading: true })
    const resp = await getSecurityQuestions()
    if (!resp) {
      this.setState({ error: 'unable to load data' })
      return
    } else if (resp.type === 'LOGIN_FAIL') {
      this.setState({ failed: !this.state.failed, modal: !this.state.modal })
      return
    }
    let questions = resp.response.data.result.questions
    if (questions) {
      this.setState({ questions: questions })
    } else {
      this.setState({ loading: false, errorMessage: resp.response.message })
    }
  }

  errorToggle = () => {
    this.setState({ modal: !this.state.modal, failed: !this.state.failed })
  }

  handlePassChange = async e => {
    e.preventDefault()
    const result = await updateSecurityQuestion(
      this.state.questionIdx,
      this.state.securityQuestionAnswer,
      this.state.oldPassword
    )
    if (
      result.error != null &&
      (result.error.response.status === 400 || result.error.response.status === 500)
    ) {
      this.setState({
        responseMessage: result.error.response.data.message
      })
      return
    }

    this.setState({ correctMessage: 'Security Question has been changed!' })
  }

  render() {
    return (
      <div>
        <Navbar className="nav-absolute" />
        <div className="background">
          <BackgroundSlideshow images={[b1, b3, b4, b5, b6]} animationDelay={5000} />
        </div>
        <Row>
          <div className="foreground">
            <Card className="interview-card">
              <CardTitle>
                <div className="text-centered" id="login-kiva-logo">
                  <img src={kivaLogo} alt="Kiva logo" />
                </div>
              </CardTitle>

              <CardBody>
                {!!this.state.questions ? (
                  <React.Fragment>
                    <div className="text-centered">
                      <Dropdown isOpen={this.state.dropdownOpen} toggle={this.toggle}>
                        <DropdownToggle caret color="transparent">
                          {this.state.questionIdx === -1
                            ? 'Security Question'
                            : this.state.questions[this.state.questionIdx]}
                        </DropdownToggle>
                        <DropdownMenu>
                          {this.state.questions.map((question, idx) => (
                            <DropdownItem onClick={this.pickDropDown.bind(null, idx)}>
                              {question}
                            </DropdownItem>
                          ))}
                        </DropdownMenu>
                      </Dropdown>
                    </div>
                    <Input
                      type="email"
                      name="email"
                      placeholder="Answer"
                      id="exampleAnswer"
                      maxLength="64"
                      pattern={EMAIL_REGEX}
                      value={this.state.securityQuestionAnswer}
                      onChange={this.handleChangeSecurityAnswer}
                      required
                    />
                  </React.Fragment>
                ) : null}
                <Form>
                  <FormGroup>
                    <Input
                      name="oldPassword"
                      placeholder="Password"
                      type="password"
                      maxLength="128"
                      id="examplePassword"
                      value={this.state.oldPassword}
                      onChange={this.handleChange}
                      required
                    />
                  </FormGroup>
                  <div className="text-centered">
                    <Button
                      color="success"
                      size="lg"
                      onClick={() => this.props.history.push('/')}
                      className="right securitybtn"
                    >
                      Back to Login
                    </Button>
                    {''}
                    <Button
                      color="success"
                      size="lg"
                      onClick={this.handlePassChange}
                      className="left left-margin-lg securitybtn"
                    >
                      Change Question
                    </Button>
                  </div>
                </Form>
                <p style={{ color: 'red', textAlign: 'center' }}>
                  {this.state.responseMessage ? this.state.responseMessage : ''}
                </p>
                <p style={{ color: 'green', textAlign: 'center' }}>
                  {this.state.correctMessage ? this.state.correctMessage : ''}
                </p>
              </CardBody>
            </Card>
          </div>
        </Row>
      </div>
    )
  }
}

export default connect()(WithAuth(ChangePassword))
