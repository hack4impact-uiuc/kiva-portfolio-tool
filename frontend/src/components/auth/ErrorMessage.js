import React, { Component } from 'react'

/**
 * If the user navigates to page that doesn't exist
 * shows them this error 404 not found page
 */
export class ErrorMessage extends Component {
  render() {
    return (
      <div>
        <h3 style={{ textAlign: 'center', padding: 150 }}>
          ERROR 404. This page could not be found. :(
        </h3>
      </div>
    )
  }
}

export default ErrorMessage
