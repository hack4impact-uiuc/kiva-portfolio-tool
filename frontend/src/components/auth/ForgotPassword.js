import { Link } from 'react-router-dom'
import React, { Component } from 'react'
import { Form, Button, FormGroup, Input, Card, Alert, CardBody, CardTitle } from 'reactstrap'
import BackgroundSlideshow from 'react-background-slideshow'

import { connect } from 'react-redux'

import Navbar from '../NavBar'

import {
  getSecurityQuestionForUser,
  submitSecurityQuestionAnswer,
  resetPassword
} from '../../utils/ApiWrapper'
import { setCookie } from '../../utils/cookie'

import kivaLogo from '../../media/kivaPlainLogo.png'
import b1 from '../../media/b1-min.jpg'
import b3 from '../../media/b3-min.jpg'
import b4 from '../../media/b4-min.jpg'
import b5 from '../../media/b5-min.jpg'
import b6 from '../../media/b6-min.jpg'

import '../../styles/index.css'
import '../../styles/login.css'

const EMAIL_REGEX =
  "([a-zA-Z0-9!#$%&'*+/=?^_`{|}~-]+)@([a-zA-Z0-9!#$%&'*+/=?^_`{|}~-]+).([a-zA-Z]{2,3}).?([a-zA-Z]{0,3})"

const mapStateToProps = state => ({
  language: state.user.language
})

/**
 * This Component handles the case where a user forgets their password
 * It contains a form to take in any information required for the user to reset their password
 * It is a multi-step recovery system
 * It has a button to return back to the login page
 */
export class ForgotPassword extends Component {
  constructor(props) {
    super(props)
    this.state = {
      email: '',
      question: '',
      errorMessage: '',
      answer: '',
      pin: '',
      password: '',
      password2: '',
      loadingAPI: false,
      submitNewPassword: false
    }
  }

  handleChange = event => {
    this.setState({ [event.target.name]: event.target.value })
  }

  handleGetSecurityQuestion = async e => {
    e.preventDefault()
    const result = await getSecurityQuestionForUser(this.state.email)

    if (result == null) {
      this.setState({
        wrongInfo: !this.state.wrongInfo,
        errorMessage: 'Wrong email!'
      })
      return
    }

    let question = result.response.data.result.question

    if (!question) {
      this.setState({
        wrongInfo: !this.state.wrongInfo,
        errorMessage: result.response.message
      })
    } else {
      this.setState({
        wrongInfo: !this.state.wrongInfo
      })
      this.setState({ question: question, errorMessage: '' })
    }
  }

  handleSubmitSecurityAnswer = async e => {
    e.preventDefault()

    this.setState({ loadingAPI: true })
    const result = await submitSecurityQuestionAnswer(this.state.email, this.state.answer)

    if (result == null) {
      this.setState({
        wrongInfo: !this.state.wrongInfo,
        errorMessage: 'Wrong answer!',
        loadingAPI: false
      })
      return
    }
    this.setState({ submitNewPassword: true, errorMessage: '' })
  }

  handleSubmitNewPassword = async e => {
    e.preventDefault()
    if (this.state.password !== this.state.password2) {
      this.setState({ errorMessage: "Passwords don't match!" })
      return
    }
    const result = await resetPassword(
      this.state.email,
      this.state.answer,
      this.state.pin,
      this.state.password
    )

    if (result == null) {
      this.setState({
        wrongInfo: !this.state.wrongInfo,
        errorMessage: 'Wrong input!'
      })
      return
    }

    let token = result.response.data.result.token

    if (!token) {
      this.setState({
        wrongInfo: !this.state.wrongInfo,
        errorMessage: result.response.message
      })
    } else {
      this.setState({
        wrongInfo: !this.state.wrongInfo
      })
      await setCookie('token', token)
      this.setState({ successfulSubmit: true })
      this.props.history.push('/')
    }
  }

  languages = {
    English: {
      pin: 'PIN',
      password: 'Password',
      confirm: 'Confirm password',
      reset: 'Reset password',
      back: 'Back to login page',
      email: 'Email',
      getSecurityQuestion: 'Get security question',
      answer: 'Answer',
      submit: 'Submit answer'
    },
    Spanish: {
      pin: 'PIN (Spanish)',
      password: 'Password (Spanish)',
      confirm: 'Confirm password (Spanish)',
      reset: 'Reset password (Spanish)',
      back: 'Back to login page (Spanish)',
      email: 'Email (Spanish)',
      getSecurityQuestion: 'Get security question (Spanish)',
      answer: 'Answer (Spanish)',
      submit: 'Submit answer (Spanish)'
    },
    French: {
      pin: 'PIN (French)',
      password: 'Password (French)',
      confirm: 'Confirm password (French)',
      reset: 'Reset password (French)',
      back: 'Back to login page (French)',
      email: 'Email (French)',
      getSecurityQuestion: 'Get security question (French)',
      answer: 'Answer (French)',
      submit: 'Submit answer (French)'
    },
    Portuguese: {
      pin: 'PIN (Portuguese)',
      password: 'Password (Portuguese)',
      confirm: 'Confirm password (Portuguese)',
      reset: 'Reset password (Portuguese)',
      back: 'Back to login page (Portuguese)',
      email: 'Email (Portuguese)',
      getSecurityQuestion: 'Get security question (Portuguese)',
      answer: 'Answer (Portuguese)',
      submit: 'Submit answer (Portuguese)'
    }
  }

  render() {
    let text = this.languages[this.props.language]
    if (!text) {
      text = this.languages['English']
    }

    return (
      <div>
        <Navbar className="nav-absolute" />
        <div className="background">
          <BackgroundSlideshow images={[b1, b3, b4, b5, b6]} animationDelay={5000} />
        </div>
        {this.state.errorMessage !== '' && <Alert color="danger">{this.state.errorMessage}</Alert>}
        {this.state.submitNewPassword ? (
          <div className="foreground">
            <Card className="interview-card">
              <CardTitle>
                <div className="text-centered" id="login-kiva-logo">
                  <img src={kivaLogo} alt="Kiva logo" />
                </div>
              </CardTitle>

              <CardBody>
                <Form>
                  <FormGroup>
                    <Input
                      name="pin"
                      placeholder={text.pin}
                      id="examplePin"
                      value={this.state.pin}
                      onChange={this.handleChange}
                      required
                    />
                  </FormGroup>
                  <FormGroup>
                    <Input
                      type="password"
                      name="password"
                      placeholder={text.password}
                      id="examplePassword"
                      minLength="8"
                      maxLength="64"
                      value={this.state.password}
                      onChange={this.handleChange}
                      required
                    />
                  </FormGroup>
                  <FormGroup>
                    <Input
                      type="password"
                      name="password2"
                      placeholder={text.confirm}
                      id="exampleConfirm"
                      minLength="8"
                      maxLength="64"
                      value={this.state.password2}
                      onChange={this.handleChange}
                      required
                    />
                  </FormGroup>
                  <div className="text-centered">
                    <Button
                      color="success"
                      size="lg"
                      onClick={this.handleSubmitNewPassword}
                      className="right"
                    >
                      {text.reset}
                    </Button>{' '}
                  </div>
                </Form>
              </CardBody>
              <div style={{ textAlign: 'center' }}>
                <Link to="/login">{text.back}</Link>
              </div>
            </Card>
          </div>
        ) : (
          <div className="foreground">
            {this.state.question === '' ? (
              <Card className="interview-card">
                <CardTitle>
                  <div className="text-centered" id="login-kiva-logo">
                    <img src={kivaLogo} alt="Kiva logo" />
                  </div>
                </CardTitle>

                <CardBody>
                  <Form>
                    <FormGroup>
                      <Input
                        type="email"
                        name="email"
                        placeholder={text.email}
                        id="exampleEmail"
                        maxLength="64"
                        pattern={EMAIL_REGEX}
                        value={this.state.email}
                        onChange={this.handleChange}
                        required
                      />
                    </FormGroup>
                    <div className="text-centered">
                      <Button
                        color="success"
                        size="lg"
                        onClick={this.handleGetSecurityQuestion}
                        className="right"
                      >
                        {text.getSecurityQuestion}
                      </Button>
                    </div>
                  </Form>
                </CardBody>
                <div style={{ textAlign: 'center' }}>
                  <Link to="/login">{text.back}</Link>
                </div>
              </Card>
            ) : (
              <Card className="interview-card">
                <CardTitle>
                  <div className="text-centered" id="login-kiva-logo">
                    <img src={kivaLogo} alt="Kiva logo" />
                  </div>
                </CardTitle>

                <CardBody>
                  <Form>
                    <FormGroup>
                      <p> {this.state.question}</p>
                      <Input
                        type="answer"
                        name="answer"
                        placeholder={text.answer}
                        id="exampleAnswer"
                        onChange={this.handleChange}
                        required
                      />
                    </FormGroup>
                    <div className="text-centered">
                      <Button
                        color="success"
                        size="lg"
                        onClick={this.handleSubmitSecurityAnswer}
                        className="right"
                        disabled={this.state.loadingAPI}
                      >
                        {text.submit}
                      </Button>
                    </div>
                  </Form>
                </CardBody>
                <div style={{ textAlign: 'center' }}>
                  <Link to="/login">{text.back}</Link>
                </div>
              </Card>
            )}
          </div>
        )}
      </div>
    )
  }
}
export default connect(mapStateToProps)(ForgotPassword)
