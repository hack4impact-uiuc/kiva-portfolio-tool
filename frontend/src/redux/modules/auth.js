const LOGIN = 'auth/login'
const LOAD = 'auth/begin_loading'
const LOAD_UPDATES = 'auth/load_updates'

const initialState = {
  verified: false
  // May need for later
  /* loading: false
  updates: {} */
}

export default function reducer(state = initialState, action) {
  switch (action.type) {
    case LOGIN:
      return {
        ...state,
        verified: action.value
      }

    // May need Load for later
    /* case LOAD:
      return {
        ...state,
        loading: action.value
      }
    case LOAD_UPDATES:
      return {
        ...state,
        updates: action.value
      } */
    default:
      return state
  }
}

export const login = value => ({
  type: LOGIN,
  value
})

// May need for later
/* export const loadUpdates = value => ({
  type: LOAD_UPDATES,
  value
})

export const beginLoading = () => ({
  type: LOAD,
  value: true
})

export const endLoading = () => ({
  type: LOAD,
  value: false
}) */
