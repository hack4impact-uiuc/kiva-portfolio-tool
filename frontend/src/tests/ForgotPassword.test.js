import React from 'react'
import { ForgotPassword } from '../components/auth/ForgotPassword'
import { Form, Button, FormGroup, Input, Card, CardBody, CardTitle } from 'reactstrap'
import Adapter from 'enzyme-adapter-react-16'
import { shallowToJson } from 'enzyme-to-json'
import { configure, shallow } from 'enzyme'
import chai from 'chai'
global.jestExpect = global.expect
global.expect = chai.expect

beforeAll(() => {
  configure({ adapter: new Adapter() })
})

test(' renders correctly', () => {
  const component = shallow(<ForgotPassword />)

  let tree = shallowToJson(component)
  jestExpect(tree).toMatchSnapshot()
})

describe('< />', () => {
  it('renders All major components', () => {
    const wrapper = shallow(<ForgotPassword />)
    expect(wrapper.find(Form)).to.have.lengthOf(1)
    expect(wrapper.find(Button)).to.have.lengthOf(1)
    expect(wrapper.find(FormGroup)).to.have.lengthOf(1)
    expect(wrapper.find(Input)).to.have.lengthOf(1)
    expect(wrapper.find(Card)).to.have.lengthOf(1)
    expect(wrapper.find(CardBody)).to.have.lengthOf(1)
    expect(wrapper.find(CardTitle)).to.have.lengthOf(1)
  })

  it('renders css', () => {
    const wrapper = shallow(<ForgotPassword />)
    expect(wrapper.find('.nav-absolute')).to.have.lengthOf(1)
    expect(wrapper.find('.background')).to.have.lengthOf(1)
    expect(wrapper.find('.foreground')).to.have.lengthOf(1)
    expect(wrapper.find('.interview-card')).to.have.lengthOf(1)
    expect(wrapper.find('.text-centered')).to.have.lengthOf(2)
  })
})
