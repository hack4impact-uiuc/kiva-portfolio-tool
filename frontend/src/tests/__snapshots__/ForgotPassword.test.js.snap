// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` renders correctly 1`] = `
<div>
  <Connect(withRouter(NavBar))
    className="nav-absolute"
  />
  <div
    className="background"
  >
    <ReactBackgroundSlideshow
      alt="background slideshow"
      animationDelay={5000}
      disableClick={false}
      disableInterval={false}
      images={
        Array [
          "b1-min.jpg",
          "b3-min.jpg",
          "b4-min.jpg",
          "b5-min.jpg",
          "b6-min.jpg",
        ]
      }
    />
  </div>
  <div
    className="foreground"
  >
    <Card
      className="interview-card"
      tag="div"
    >
      <CardTitle
        tag="div"
      >
        <div
          className="text-centered"
          id="login-kiva-logo"
        >
          <img
            alt="Kiva logo"
            src="kivaPlainLogo.png"
          />
        </div>
      </CardTitle>
      <CardBody
        tag="div"
      >
        <Form
          tag="form"
        >
          <FormGroup
            tag="div"
          >
            <Input
              id="exampleEmail"
              maxLength="64"
              name="email"
              onChange={[Function]}
              pattern="([a-zA-Z0-9!#$%&'*+/=?^_\`{|}~-]+)@([a-zA-Z0-9!#$%&'*+/=?^_\`{|}~-]+).([a-zA-Z]{2,3}).?([a-zA-Z]{0,3})"
              placeholder="Email"
              required={true}
              type="email"
              value=""
            />
          </FormGroup>
          <div
            className="text-centered"
          >
            <Button
              className="right"
              color="success"
              onClick={[Function]}
              size="lg"
              tag="button"
            >
              Get security question
            </Button>
          </div>
        </Form>
      </CardBody>
      <div
        style={
          Object {
            "textAlign": "center",
          }
        }
      >
        <Link
          href="/login"
          prefetch={true}
          replace={false}
          to="/login"
        >
          <a
            href="/login"
          >
            Back to login page
          </a>
        </Link>
      </div>
    </Card>
  </div>
</div>
`;
